src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -448,6 +448,14 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             var hasErrors = localSymbol.ScopeBinder
                 .ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
+            foreach (var p in node.ParameterList.Parameters)
+            {
+                if (p.IsArgList)
+                {
+                    diagnostics.Add(ErrorCode.ERR_IllegalVarArgs, p.Location);
```

Comment:
I was going to eagerly report diagnostics that don't have any risk of cycles and only put lazy things into GetDeclarationDiagnostics. Would you prefer that everything be reported in GetDeclarationDiagnostics if feasible?